contrib/runners/mistral_v2/query/mistral_v2.py

```diff
@@ -134,15 +135,15 @@ def _determine_execution_status(self, execution_id, wf_state, tasks):
         # Identify the list of tasks that are not in completed states.
         active_tasks = [t for t in tasks if t['state'] not in DONE_STATES]
 
-        # On cancellation, mistral workflow executions are paused so that tasks
-        # can gracefully reach completion. This is only temporary until a canceled
-        # status is added to mistral.
-        if (wf_state in DONE_STATES or wf_state == 'PAUSED') and is_action_canceled:
-            status = action_constants.LIVEACTION_STATUS_CANCELED
-        elif wf_state in DONE_STATES and not is_action_canceled and active_tasks:
-            status = action_constants.LIVEACTION_STATUS_RUNNING
+        # Keep the execution in running state if there are active tasks.
+        # In certain use cases, Mistral sets the workflow state to
+        # completion prior to task completion.
+        if is_action_canceled and active_tasks:
+            status = action_constants.LIVEACTION_STATUS_CANCELING
         elif wf_state not in DONE_STATES:
             status = action_constants.LIVEACTION_STATUS_RUNNING
+        elif wf_state in DONE_STATES and active_tasks:
+            status = action_constants.LIVEACTION_STATUS_RUNNING
```

Comment:
Having a cancel with force options sounds good. I'll add this feature in a separate PR.